providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/GrpcConnector.java

```diff
@@ -37,7 +38,7 @@ public class GrpcConnector {
     private final long deadline;
 
     private final Cache cache;
-    private final BiConsumer<Boolean, List<String>> stateConsumer;
+    private final TriConsumer<Boolean, List<String>, Map<String, Object>> onConnectionEvent;
```

Comment:
is maybe now the time, to use a pojo for this and a normal consumer?

Reply:
you mean a pojo for the metadata map, or a proper named functional interface for the TriConsumer?

Reply:
a pojo for boolean, list and map

Reply:
We may actually already have such a POJO :sweat_smile: 

I think I'll make this change.

Reply:
see also https://github.com/open-feature/java-sdk-contrib/pull/925#discussion_r1743094379 - from not so long ago

Reply:
Done: https://github.com/open-feature/java-sdk-contrib/pull/967/commits/667946fba576537e134f87ba85e10c2c77f11f66